federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPUtil.java

```diff
@@ -129,24 +129,35 @@ public static String convertObjectGUIToByteString(byte[] objectGUID) {
     public static String decodeObjectGUID(byte[] objectGUID) {
         StringBuilder displayStr = new StringBuilder();
 
-        displayStr.append(convertToDashedString(objectGUID));
+        byte[] withLittleEndian = new byte[] { objectGUID[3], objectGUID[2], objectGUID[1], objectGUID[0],
```

Comment:
Is it possible to avoid those changes and also avoid the changes in "convertToDashedString" method and instead add the changes to the "decodeGuid" method and ensure that it shuffles bytes to correct order before calling "convertToDashedString" method? Just want to avoid changes in this PR, which would potentially mean breaking MSAD (that one is far more important for us than Novell eDirectory TBH).

Also is it possible to add javadoc to method "decodeGuid" that it is used typically for novell eDirectory?

Reply:
Sure! I have updated the PR. Please check again.

btw: Do we need to worry about backward compatibility for existing Keycloak instances with eDirectory connectivity?